internal/backends/python/python.go

```diff
@@ -347,7 +333,176 @@ func makePythonPoetryBackend(python string) api.LanguageBackend {
 	}
 }
 
-func listSpecfile() (map[api.PkgName]api.PkgSpec, error) {
+func ensureVenvExists(python string, venvPath string) (string, string) {
+	// Create venv if it does not exsit
+	if !util.Exists(venvPath) {
+		util.GetCmdOutputFallible([]string{
+			python, "-m", "venv", venvPath,
+		})
+	}
+	pipPath := path.Join(venvPath, "bin", "pip")
+	return venvPath, pipPath
+}
+
+// makePythonPipBackend returns a backend for invoking poetry, given an arg0 for invoking Python
+// (either a full path or just a name like "python3") to use when invoking Python.
+func makePythonPipBackend(python string) api.LanguageBackend {
+	var pipFlags []PipFlag
+
+	b := api.LanguageBackend{
+		Name:                 "python3-pip",
+		Specfile:             "requirements.txt",
+		Alias:                "python-python3-pip",
+		FilenamePatterns:     []string{"*.py"},
+		Quirks:               api.QuirksAddRemoveAlsoInstalls | api.QuirksCannotLock | api.QuirksNotReproducible,
```

Comment:
Yeah, I could definitely see that. The reason I didn't go that way was because it would be defining yet another bespoke standard. If we're guaranteed to be running inside Conda, perhaps relying on Conda to do locking makes sense, but then that just adds more interdependency spaghetti.

Reply:
Resolved over Slack. Lockfiles should provide guarantees that are outside of the scope of what `requirements.txt` offers. Perhaps https://discuss.python.org/t/what-is-the-status-on-a-pip-lock-file/6669/8 will resolve this someday.